packages/auth-plugins/src/UserUtils.ts

```diff
@@ -0,0 +1,74 @@
+import { User, UserPermissions } from '@deephaven/redux';
+
+/**
+ * Retrieve a value from the AppInit config
+ * @param serverConfig Server config map
+ * @param key The AppInit key to retrieve
+ * @returns The value for the AppInit key
+ */
+export function getAppInitValue(
+  serverConfig: Map<string, string>,
+  key: string
+): string | undefined {
+  return serverConfig.get(`internal.webClient.appInit.${key}`);
+}
+
+/**
+ * Retrieve a user object provided the server config and overrides
+ * @param serverConfig Server config map
+ * @param overrides Override values for the user
+ * @param permissionsOverrides Override specific permissions for the user
+ * @returns The user object
+ */
+export function getUserFromConfig(
+  serverConfig: Map<string, string>,
+  overrides: Partial<Omit<User, 'permissions'>> = {},
+  permissionsOverrides: Partial<UserPermissions> = {}
+): User {
+  function getValue(key: string): string | undefined {
+    return getAppInitValue(serverConfig, key);
+  }
+  function getBooleanValue(key: string, defaultValue = true): boolean {
```

Comment:
Maybe require the default value here? You provide it explicitly everywhere anyway